tests/test_scripts.py

```diff
@@ -0,0 +1,310 @@
+        ]
+    },
+    'squeeze_urban_links.py': {
+        'args': [
+            '--network', EXAMPLE_NETWORK,
+            '--projection', PROJECTION,
+            '--study_area', os.path.join(EXAMPLE_DATA_DIR, 'Fitzrovia_polygon.geojson'),
+            '--urban_geometries', os.path.join(EXAMPLE_DATA_DIR, 'Fitzrovia_urban_polygon.geojson'),
+            '--freespeed', '0.5',
+            '--capacity', '0.5',
+
+        ],
+        'expected_files': [
+            'network.xml',
+            os.path.join('supporting_outputs', 'capacity_before.geojson'),
+            os.path.join('supporting_outputs', 'freespeed_after.geojson'),
+            os.path.join('supporting_outputs', 'urban_network_links.geojson'),
+            os.path.join('supporting_outputs', 'capacity_after.geojson'),
+            os.path.join('supporting_outputs', 'freespeed_before.geojson')
+        ]
+    },
+    'validate_network.py': {
+        'args': [
+            '--network', EXAMPLE_NETWORK,
+            '--schedule', EXAMPLE_SCHEDULE,
+            '--vehicles', EXAMPLE_VEHICLES,
+            '--projection', PROJECTION,
+        ],
+        'expected_files': [
+            'validation_report.json'
+        ]
+    }
+}
+
+
+def run_process(command):
+    process = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                               universal_newlines=True)
+    lines = []
+    while process.poll() is None:
+        line = process.stderr.readline()
+        lines.append(line)
+    exit_code = process.poll()
+    console_output = '\t' + "\t".join(lines)
+    return exit_code, console_output
+
+
+def check_output_files_do_not_exist(output_dir, files: list):
+    for file in files:
+        file_path = os.path.join(output_dir, file)
+        assert not os.path.exists(file_path), f"File {file_path} exists but is not expected."
+
+
+def check_output_files_exist(output_dir, files: list):
+    for file in files:
+        file_path = os.path.join(output_dir, file)
+        assert os.path.exists(file_path), f"File {file_path} does not exist but is expected."
+
+
+@pytest.mark.parametrize(
+    "script", list(SCRIPTS.keys())
+)
+def test_integration_of_example_scripts(tmpdir, script):
+    output_folder = os.path.join(tmpdir, f'script_{script.replace(".py", "")}')
+    check_output_files_do_not_exist(output_folder, SCRIPTS[script]['expected_files'])
+    command = f"python {os.path.join(os.path.dirname(__file__), '..', 'scripts', script)} "
+    command = command + ' '.join(SCRIPTS[script]['args']) + ' --output_dir ' + output_folder
+
+    exit_code, console_output = run_process(command)
+
+    if exit_code != 0:
+        raise AssertionError(f'Script {script} did not run successfully. Error message: \n\n{console_output}')
+    else:
+        check_output_files_exist(output_folder, SCRIPTS[script]['expected_files'])
+
+
+def test_google_api_script_throws_error_without_api_key(tmpdir):
```

Comment:
could we have a test that runs the script with an API key that's stored in the repository secrets? If we make one request per CI run, would that keep us easily below some threshold for being charged for the requests?

Reply:
so, we could, but I don't want to actually ping the api every time we run tests (it would also require people to have those secrets set up locally if they want to run tests or be authenticated to the aws account). There isn't a free amount of queries btw, I don't think it would break the bank tbh. There are unit tests that mock querying the api and processing the results, I really think that testing this script breaking on t'hreshold limit of requests exceeded' before querying the api is more valuable for us here.

That said, I think in the cli version in your branch, we might be able to mock this and run without pinging the api.